packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,514 @@
+ * accelerate confirmation or to cancel a previous transaction.
+ *
+ * The class supports both transaction acceleration and cancellation, with configurable
+ * fee rates based on network conditions and user-defined urgency levels.
+ *
+ * Key Features:
+ * - Supports PSBTv2 and PSBTv0 formats
+ * - Implements BIP125 rules for RBF
+ * - Allows fee subtraction from specific outputs or proportional distribution
+ * - Supports additional input selection for fee coverage
+ * - Provides transaction cancellation functionality
+ *
+ * Usage:
+ * const rbfTx = new RbfTransaction(psbt, feeMarket, options, network);
+ * const acceleratedPsbt = rbfTx.prepareAccelerated();
+ * // or
+ * const canceledPsbt = rbfTx.prepareCanceled(destinationAddress);
+ */
+export class RbfTransaction {
+  private readonly originalPsbt: PsbtV2;
+  private newPsbt: PsbtV2;
+  private readonly network: Network;
+  private _targetFeeRate: FeeRateSatsPerVByte;
+  private readonly feeOutputIndex?: number;
+  private readonly dustThreshold: BigNumber;
+  private readonly additionalUtxos: UTXO[];
+  private readonly requiredSigners: number;
+  private readonly totalSigners: number;
+  private readonly changeOutputIndices: number[];
+
+  /**
+   * Constructor for RbfTransaction
+   *
+   * @param {RbfTransactionOptions} options - Configuration options for the RBF transaction
+   * @throws {Error} If the PSBT is not in a valid format or not signaling RBF
+   */
+  constructor(options: RbfTransactionOptions) {
+    this.originalPsbt = initializePsbt(options.psbt);
+    this.newPsbt = new PsbtV2();
+    this.originalPsbt.copy(this.newPsbt);
+    this.network = options.network;
+    this._targetFeeRate = options.targetFeeRate;
+    this.feeOutputIndex = options.feeOutputIndex;
+    this.dustThreshold = new BigNumber(
+      options.dustThreshold || DEFAULT_DUST_THRESHOLD,
+    );
+    this.additionalUtxos = options.additionalUtxos || [];
+    this.requiredSigners = options.requiredSigners;
+    this.totalSigners = options.totalSigners;
+    this.changeOutputIndices = options.changeOutputIndices;
+    this.validatePsbt();
+  }
+
+  private validatePsbt(): void {
+    if (!this.newPsbt.isRBFSignaled) {
+      throw new Error("This transaction is not signaling RBF.");
+    }
+  }
+
+  // getters
+
+  get psbt(): PsbtV2 {
+    return this.newPsbt;
+  }
+
+  get targetFeeRate(): FeeRateSatsPerVByte {
+    return this._targetFeeRate;
+  }
+
+  get totalInputValue(): string {
+    return calculateTotalInputValue(this.newPsbt).toString();
+  }
+
+  get totalOutputValue(): string {
+    return calculateTotalOutputValue(this.newPsbt).toString();
+  }
+
+  get currentFee(): string {
+    return new BigNumber(this.totalInputValue)
+      .minus(this.totalOutputValue)
+      .toString();
+  }
+
+  get requiredFee(): string {
+    return new BigNumber(this.estimateVsize())
+      .multipliedBy(this.targetFeeRate)
+      .toString();
+  }
```

Comment:
This also isn't right. You need to look at the RBF rules here. If you broadcast a tx at 5 sats/vbyte and it gets stuck because current median fees are 70 sats/vbyte, it's not enough to just update to have 70 sats/vbyte as this will be rejected by nodes under certain conditions. There are anti-ddos rules for RBF to avoid spamming nodes which require you, for example, to have to pay enough fees to cover the cost of the transaction you are replacing as well. So if the target is 70 sats/vbyte, the actual required fee rate will likely be higher. The user/consumer of this function will likely just be looking at market fee rate conditions, so this class needs to help determine what the fees and fee rate will need to be to replace given those market conditions. 

Also I think this is better for clarity:
```suggestion
  get requiredAbsoluteFee(): string {
    return new BigNumber(this.estimateVsize())
      .multipliedBy(this.targetFeeRate)
      .toString();
  }
```

Reply:
Took a lot of references for this , for BIP-125 and then various articles including :

1) (stack-over flow)[https://bitcoin.stackexchange.com/questions/119325/how-to-properly-calculate-fee-for-rbf-transaction#:~:text=Since%20the%20incrementalfee%20is%201,pay%20at%20least%202000%20%E1%B9%A9.]

2) [rbf-improvements ] (https://gist.github.com/glozow/25d9662c52453bd08b4b4b1d3783b9ff)

3) [incremental Relay Fee] (https://bitcoin.stackexchange.com/questions/120778/what-is-incremental-relay-feerate)

created a new replacement method named [calculateFeeIncrease](https://github.com/caravan-bitcoin/caravan/pull/114/commits/4f4f97a08243f63ca31c4f0fc34216671e97e1de#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR163) based on above references ...

so the `requiredFee` method now looks like [this](https://github.com/caravan-bitcoin/caravan/pull/114/commits/ee553e733d062d4b7913ca7c62198f992174b4cf#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR148) 

```js
  get requiredAbsoluteFee(): string {
    return this.calculateFeeIncrease()
      .plus(new BigNumber(this.currentFee))
      .toString();
  }
```

Reply:
This class is depreciated now